tests/integration/test_charm.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python3
+# Copyright 2024 Tony Meyer
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./charmcraft.yaml").read_text())
+APP_NAME = METADATA["name"]
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    """Build the charm-under-test and deploy it together with related charms.
+
+    Assert on the unit status before any relations/configurations take place.
+    """
+    # Build and deploy charm from local source folder
+    charm = await ops_test.build_charm(".")
+    resources = {"takahe-image": METADATA["resources"]["takahe-image"]["upstream-source"]}
+
+    # Deploy the charm and wait for blocked/idle status
+    await asyncio.gather(
```

Comment:
This test is the one that's provided when you do `charmcraft init --profile=kubenetes` (except for the resource name). I haven't got to integration tests yet (although should).

In general, I feel there's way too much async in pytest_operator, because most of the time you need to do things synchronously anyway (e.g. the build needs to be finished first, then the deploy has to finish first). I'm somewhat reserving judgement on that until I've written more complicated tests.

I'm not sure why the template test is written this way - maybe it's to show an example of how you can do this? I agree, there's no way that the app will be idle until well after deploy is done. I'll look at adjusting this when I properly write integration tests.